routes/presenceRouter.js

```diff
@@ -0,0 +1,132 @@
+    }
+}
+
+/**
+ * Get the next trainings of a team
+ * Please don't ask me how this query works, I don't know either
+ * @param teamId the id of the team
+ */
+function getNextTrainings(teamId) {
+    return pool.query(` WITH times AS (SELECT CAST(EXTRACT(epoch FROM day AT TIME ZONE 'CEST') AS INTEGER) AS time_series,
+                        to_char(to_timestamp(EXTRACT(epoch FROM day AT TIME ZONE 'CEST')), 'DD.MM.YYYY') AS readable_date 
+                        FROM generate_series(current_date, current_date + interval '14 days', '1 day') AS day),
+                         
+                        tmpData AS (
+                            SELECT times.time_series, readable_date, account.username, presence.state, CASE WHEN presence.from IS NULL THEN '00:00' ELSE presence.from END AS from, 
+                            CASE WHEN presence.until IS NULL THEN '23:59' ELSE presence.until END AS until FROM account 
+                            LEFT JOIN teammembership ON teammembership.account_fk = account.id CROSS JOIN times 
+                            LEFT JOIN presence ON presence.account_fk = account.id AND presence.date = times.time_series AND presence.state <> 2 
+                            WHERE teammembership.team_fk = $1 AND presence.date > EXTRACT(epoch FROM (CURRENT_DATE - INTERVAL '1 day' + INTERVAL '23 hours 59 minutes')::timestamp AT TIME ZONE 'CEST') ORDER BY time_series),
+                            
+                        counts AS (
+                            SELECT time_series, COUNT(*) AS num_rows FROM tmpData GROUP BY time_series),
+                            
+                        tmpData2 AS(
+                            SELECT tmpData.* FROM tmpData JOIN counts ON tmpData.time_series = counts.time_series
+                            WHERE num_rows = (  SELECT COUNT(*) FROM account LEFT JOIN teammembership ON teammembership.account_fk = account.id 
+                                                WHERE teammembership.team_fk=$1) ORDER BY tmpData.time_series),
+                                                
+                        tmpData3 AS (
+                            SELECT *, (SELECT MAX(x.from) FROM tmpData2 AS x WHERE x.time_series = tmpData2.time_series) AS starttime,
+                            (SELECT MIN(x.until) FROM tmpData2 AS x WHERE x.time_series = tmpData2.time_series) AS endtime FROM tmpData2),
+                        tmpData4 AS (
+                            SELECT *, CASE WHEN (SELECT COUNT(*) FROM tmpData3 AS x WHERE x.time_series = tmpData3.time_series AND x.state = 3) > 0 
+                            THEN 'unsure' ELSE 'sure'  END AS trainingtype FROM tmpData3 WHERE starttime < endtime)
+                            
+                        SELECT readable_date, starttime, endtime, trainingtype FROM tmpData4
+                        GROUP BY readable_date, starttime, endtime, trainingtype
+                        ORDER BY readable_date`, [teamId]);
+}
```

Comment:
'tis a large query. I think it would make sense to add a comment what this query exactly does. And yes I've read the javadoc above.

Reply:
Added some comments